### DISCUSSION TITLE
How to read/peek at capsule state without "subscribing" to rebuilds

### DISCUSSION THREAD

**Discussion by Kypsis** [GregoryConrad/rearch-dart#108] [Q&A] [answered]
How to read/peek at capsule state within a widget or another capsule without "subscribing" to rebuilds. I hope this question even makes sense in context of Rearch.

**Comment by GregoryConrad** [comment, marked as answer]
"Reading/peeking" instead of "subscribing" doesn't normally make sense in the context of ReArch because it favors more imperative patterns rather than reactive/declarative. Would you mind explaining what you are trying to do?

**Reply by Kypsis** [reply]
This question was mainly in the context of [https://github.com/GregoryConrad/rearch-dart/discussions/109](https://github.com/GregoryConrad/rearch-dart/discussions/109) but you already provided an example of an "escape hatch" with `container.read` there.
